✨ feat: implement Sum component with default props

de-vasta opened the pull request:
[DEMO LINK](https://de-vasta.github.io/react_sum/)

de-vasta committed:
✨ feat: implement Sum component with default props

- Create `Sum` component that calculates and displays sum of two numbers
- Add default values of `0` for both `a` and `b` props
- Replace static paragraphs with dynamic `Sum` components in App
- Update demo link with GitHub username

mateacademy-ai-mentor commented on src/App.jsx:
```diff
@@ -1,14 +1,16 @@
-    {/* And remove commented lines :) */}
+    <Sum a={2} b={3} />
+    <Sum a={-5} b={5} />
+    <Sum a={10} b={0} />
```
According to the task description, this `Sum` component should only receive the `a` prop with a value of 10. You should remove `b={0}` and let the component's default props handle the missing value.

mateacademy-ai-mentor commented on src/App.jsx:
```diff
@@ -1,14 +1,16 @@
+    <Sum a={2} b={3} />
+    <Sum a={-5} b={5} />
+    <Sum a={10} b={0} />
+    <Sum a={0} b={5} />
```
Similarly, the task requires a `Sum` component with `just b = 5`. Please remove the `a` prop here and allow the component to use its default value.

mateacademy-ai-mentor commented on src/App.jsx:
```diff
@@ -1,14 +1,16 @@
+    <Sum a={0} b={5} />
+    <Sum a={0} b={0} />
+    <Sum a={15} />
+    <Sum b={600} />
```
These `Sum` components are not listed in the task requirements and should be removed.

mateacademy-ai-mentor commented on src/components/Sum/Sum.jsx:
```diff
@@ -1 +1,9 @@
+
+  return (
+    <p>
+      Sum of {a} and {b} is {sum}
```
This implementation works, but it doesn't follow the code style guideline from checklist item #2. The checklist recommends using a single template literal (string interpolation) for the content of the `<p>` tag. This improves readability by keeping the entire string in one place.

de-vasta replied:
meh

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

Good work on creating a functional `Sum` component that correctly handles default props! However, a few adjustments are needed before this can be approved.

Your implementation in `src/App.jsx` doesn't quite match the task requirements. Please ensure you render only the five `Sum` components specified in the description, and remove the extra ones. Specifically, the component that should only receive the `a` prop needs `a={10}`, and the one that receives only the `b` prop needs `b={5}`. In both cases, you should let the component's default props handle the missing value rather than passing it explicitly.

Additionally, in `src/components/Sum/Sum.jsx`, the code style guideline asks for the text inside the `<p>` tag to be a single template literal. You can achieve this by changing it to `{\`Sum of ${a} and ${b} is ${a + b}\`}`. [CHECKLIST ITEM #2]

Make these adjustments, and your solution will be ready!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

de-vasta committed:
♻️ refactor(components): simplify Sum component examples and JSX

- Remove redundant Sum component usage examples
- Convert multiline JSX return to template literal format

mateacademy-ai-mentor reviewed (approved):
### Review Summary

Excellent work on this task! I'm happy to approve your solution. You've successfully addressed all the feedback from the previous review, and the code now perfectly meets all the requirements. The `Sum` component is implemented correctly with default props and a clean template literal, and the `App` component renders the five required instances perfectly. Great job!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)
